external/CMakeLists.txt

```
@@ -262,6 +262,23 @@ if(APPLE)
 		message(FATAL_ERROR "Cannot locate OpenSSL directory\nPlease run: brew install openssl")
 	endif()
 endif()
+
+
+find_package(CURL QUIET)
+if(NOT CURL_FOUND)
+	message("System curl not found, compiling curl from source")
+	include(FetchContent)
+	FetchContent_Declare(curl
+		URL https://github.com/curl/curl/releases/download/curl-8_1_2/curl-8.1.2.tar.xz
+	)
+	FetchContent_MakeAvailable(curl)
+	target_link_libraries(libcurl INTERFACE curl)
```

```suggestion
	message("System curl not found, compiling curl from source")
	option(HTTP_ONLY "disables all protocols except HTTP (This overrides all CURL_DISABLE_* options)" ON)
	include(FetchContent)
	FetchContent_Declare(curl
		URL https://github.com/curl/curl/releases/download/curl-8_1_2/curl-8.1.2.tar.xz
	)
	FetchContent_MakeAvailable(curl)
```

We got no need for other protocols (IMAP, LDAP...)
Also the curl CMake defines libcurl and not curl.

---

Also put the curl code after the openssl one in this file.

---

Also it's better to use the git module as it won't update the file timestamps every time cmake is run.